inference-engine/thirdparty/clDNN/src/graph_optimizer/oooq_memory_dependencies.cpp

```diff
@@ -92,15 +92,26 @@ void oooq_memory_dependencies::run(program_impl& p) {
     // every node has a bit array assigned to it
     // users or the node are marked with 1 bit in this array
     std::vector<bits_64> user_bitmap(num_nodes, bits_64(num_nodes));
+    std::vector<program_node*> suspect_nodes;
```

Comment:
Is my assumption correct that the fact that node pointer is inserted into this vector serves just as a boolean sign that this node has non-constant dependencies?

If yes, then almost every node of the topology will be inserted into this vector. So, why then don't use the bitmap similar to that on the line above, and instead of creating extra loop, just test this bitmap in the existing dependency checking loop at line 162?

There are several reasons for doing so:
1) Vector insertion operation is quite slow. Bitmap is about 100x faster.
2) Bitmap will use ~64x less memory
3) No need to create a new loop - which also will be more cache local and faster

For topologies having 1000+ nodes the usage of this vector will be a significant slowdown and memory pressure factor for this pass.